### ISSUE TITLE
Wildcard rooted at a NuGet-package-defined property expands drive root

### ISSUE THREAD

**Issue by kingerja** [dotnet/msbuild#4091]
### Steps to reproduce

Project file
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup >
    <MyCopyItem Include="$(PkgEnterpriseLibrary_TransientFaultHandling_Core)\**\*.*" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="EnterpriseLibrary.TransientFaultHandling.Core" Version="1.0.0" GeneratePathProperty="true"/>
  </ItemGroup>
</Project>
```

Directory contents:
```
/
- foo.csproj
- emptyclass.cs
```

Command line
```
msbuild /t:restore foo.csproj
```
### Expected  behavior
packages restored under a second.  
Or perhaps a warning saying `hey don't be stupid $(PkgEnterpriseLibrary_TransientFaultHandling_Core) is not defined yet and \** will traverse your hard drive and take forever`.   This of course makes the first nuget restore experience horrible (which in the lab environment is every build).

Basically I think this scenario will become much more common as people switch to PackageReference items and either warning the user (which they can suppress if they don't want) or making a better decision about the context would be a more desirable user experience.

### Actual behavior
msbuild hangs while the ** glob expands across my huge system drive making restore take a very long time.

### Environment data
`msbuild /version` output:
15.9.21.664
OS info:
Version 1809 (OS Build 17763.253)

If applicable, version of the tool that invokes MSBuild (Visual Studio, dotnet CLI, etc):

This issue #3642 appears related.

**Comment by rainersigwald** [comment]
I know we've talked about this before but I also can't find the canonical bug for it. #3204 is also related.

**Comment by jeffkl** [comment]
We've worked around it in a few places by setting the env var `MsBuildSkipEagerWildCardEvaluationRegexes`.  I think we need to make that feature flag a global property so it can be set during restore by NuGet or our `/restore` functionality.

**Comment by kingerja** [comment]
This workaround MsBuildSkipEagerWildCardEvaluationRegexes was not completely sufficient on our build agents.  Nor was it sufficient to condition the item.  I am not sure of the reason why though but if you want to go with the skip eager solution I will need to dig into why it wasn't sufficient on some agents.

**Comment by rainersigwald** [comment]
Filed NuGet/Home#8871 to make the NuGet property-generation behavior easier to handle "less incorrectly".

**Comment by BenVillalobos** [comment]
Related: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1172403

**Comment by ladipro** [comment]
@kingerja, as of #5669, adding a condition to the item definition prevents the unwanted glob expansion:

```xml
<MyCopyItem Include="$(PkgEnterpriseLibrary_TransientFaultHandling_Core)\**\*.*" Condition="'$(PkgEnterpriseLibrary_TransientFaultHandling_Core)' != ''" />
```
